### PR TITLE
Multimodal projection layer adapter fix for PP>1

### PR DESCRIPTION
**Pull request by paul-gibbons**
# What does this PR do ?

When loading a non-strict dist checkpoint in NeMo it hangs at https://github.com/NVIDIA/NeMo/blob/d7bb40364c17bf322004539f851cc83df4c4c2b7/nemo/utils/callbacks/dist_ckpt_io.py#L272 at PP_rank=0. 

When loading multimodal projection layer, there will be ranks loading nothing which leads to the hang because we only load weights for rank0. Setting **validate_access_integrity** resolves this hang.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/utils/callbacks/dist_ckpt_io.py [hunk 1]
```diff
@@ -270,7 +270,10 @@ def load_checkpoint(
             sharded_state_dict = self.adjust_non_strict_load(path, sharded_state_dict)
 
         return dist_checkpointing.load(
-            sharded_state_dict=sharded_state_dict, checkpoint_dir=path, sharded_strategy=sharded_strategy
+            sharded_state_dict=sharded_state_dict,
+            checkpoint_dir=path,
+            sharded_strategy=sharded_strategy,
+            validate_access_integrity=False,
```
Comment (mikolajblaz) [inline nemo/utils/callbacks/dist_ckpt_io.py:276]: @paul-gibbons We can't turn it off always, only when absolutely necessary because of irregular load.
Can you parametrize it?

**Review by mikolajblaz — DISMISSED** [review]
Ok from my side

File: nemo/core/connectors/save_restore_connector.py [hunk 2]
```diff
@@ -253,7 +255,13 @@ def restore_from(
         # Get path where the command is executed - the artifacts will be "retrieved" there
         # (original .nemo behavior)
         loaded_params = self.load_config_and_state_dict(
-            calling_cls, restore_path, override_config_path, map_location, strict, return_config, trainer,
+            calling_cls,
```
Comment (mikolajblaz) [inline nemo/core/connectors/save_restore_connector.py:258]: Should `validate_access_integrity` be passed here?

**Review by mikolajblaz — APPROVED** [review]

**Review by yaoyu-33 — APPROVED** [review]

**Review by yaoyu-33 — APPROVED** [review]
